pkg/probe/http/http.go

```diff
@@ -113,8 +115,12 @@ func DoHTTPProbe(req *http.Request, client GetHTTPInterface) (probe.Result, stri
 			klog.V(4).Infof("Probe terminated redirects for %s, Response: %v", url.String(), *res)
 			return probe.Warning, fmt.Sprintf("Probe terminated redirects, Response body: %v", body), nil
 		}
-		klog.V(4).Infof("Probe succeeded for %s, Response: %v", url.String(), *res)
-		return probe.Success, body, nil
+		if limitReached {
+			return probe.Warning, fmt.Sprintf("Non fatal body truncation for %s, Response: %v", url.String(), body), nil
```

Comment:
I'm wondering how this would fix [the issue ](https://github.com/kubernetes/kubernetes/issues/122948)? The only difference I can see in the code that returning `probe.Warning` would [trigger ContainerProbeWarning event](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/prober.go#L113), but [probe would still succeed](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/prober.go#L118).

Reply:
@bart0sh it is not intend to fix the "issue", as i said in my commit, it is a non breaking change, i interpreted @aojea s comment in the issue as a breaking change would not be an option, thus a warning would be a compromise

Reply:
yeah, things that worked before should keep working, otherwise workloads that worked in version N , will break in N+1 if the probe logic changes

Reply:
I'm not against the change. Just trying to understand how it can help with the issue. The only outcome of this I can see is the event, but I still don't understand how that helps the reader to handle partial probe body better.

Reply:
@bart0sh I removed the body.